Assets/Scripts/Controllers/projectileController.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+public class projectileController : MonoBehaviour
+{
+    private Rigidbody2D _projectileCollision;
+    private Rigidbody2D _playerPhysicsBody;
+    private Vector2 _bulletDirection;
+    private float _bulletSpeed = 800f;
+    private Camera _cam;
+    private GameObject _playerCharacter;
+    
+    void Awake ()
+    {
+        //Get position and physics components of projectile; initialize camera variable for converting world to screen coordinates.
+        _projectileCollision = GetComponent<Rigidbody2D>();
+        _cam = Camera.main;
+
+        //Get velocity of player character.
+        _playerCharacter = GameObject.Find("Player");
+        _playerPhysicsBody = _playerCharacter.GetComponent<Rigidbody2D>();
+
+        //Get the direction of the velocity vector of the bullet.
+        _bulletDirection = Input.mousePosition - _cam.WorldToScreenPoint(_projectileCollision.position);
```

Comment:
Rather than making a bullet responsible for determining it's own direction, we should allow the shooter to determine this so that the projectile doesn't need to be coupled to the scene camera and player character.

In `CharacterController.cs`:
```cs
_projectileSpawnModifier = Input.mousePosition - _cam.WorldToScreenPoint(_rb.position); // Get thoe shoot direction
...
GameObject newProjectile = Instantiate(_weaponProjectile, _rb.position + _projectileSpawnModifier, Quaternion.identity);
newProjectile.forward = _projectileSpawnModifier.Normalized(); // A neat trick for setting the forward direction of a game object.
```

If the `CharacterController.cs` sets the direction of the transform, the entire `Awake` method is replaced with just this:
```cs
_projectileCollision = GetComponent<Rigidbody2D>();
_projectileCollision.AddForce(transform.forward * _bulletSpeed + _playerPhysicsBody.velocity);
```